Feature 1127

IKP1048576 opened the pull request:
決定木の実装
2次元的なデータに対しての決定木

Commit:
2次元データの決定木

Commit:
決定木の実装

tus-os-2023 commented:
ありがとうございます。
以下、コメントになります。

概ね使い方については問題ありません。

データの確認
↓
学習用と評価用に分割
↓
モデルごとのインスタンスの作成
↓
学習
↓
評価

この流れがわかっているかと思われます。
また、 `DecisionTreeClassifier` クラスの定義方法も最低限の理解ができているので素晴らしいです。

ここからは改善事項になります。

1. notebooks ディレクトリ直下に入れるのはやめておきましょうか。
    notebooks の直下は、あくまでこのプロジェクトのデータに対して分析を行ったものだけを管理したいです。
    `notebooks/kenpi_practice` のようなフォルダを作ってその下にこのファイルを置きましょう。
2. 説明変数の表記は慣習に則って、大文字で `X` としましょう。
    これ以降の `x_train`, `x_test` も同様で `X_train`, `X_test` としておきましょう。
3. 目的変数の表記も `t` ではなく、こちらは小文字の `y` を使いましょう。

とりあえずはこれで大丈夫かと思います。
ご対応よろしくお願い致します。

Commit:
1.保存場所の変更
notebooks直下でなく、notebooks→kenpi_practice→当notebook
2.説明変数の表記の変更
x→X
x_train→X_train
x_test→X_test
3.目的変数の表記の変更
t→y

IKP1048576 commented:
1.保存場所の変更
notebooks直下でなく、notebooks→kenpi_practice→当notebook
2.説明変数の表記の変更
x→X
x_train→X_train
x_test→X_test
3.目的変数の表記の変更
t→y
